lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -291,5 +293,12 @@ async fn _do_recover_from_snapshot(
     // Remove tmp collection dir
     tokio::fs::remove_dir_all(&tmp_collection_dir).await?;
 
+    // Remove snapshot after recovery if downloaded
+    if let Some(path) = snapshot_temp_path {
+        if let Err(err) = path.delete() {
+            log::error!("Failed to remove downloaded collection snapshot after recovery: {err}");
+        }
+    }
+
```

Comment:
Explicit deletion here so we can do some error reporting it it somehow failed. Figured this was nice, but we can decide to remove it.

Note that if `delete()` isn't called, the file is still deleted because it implements `Drop`.